string serial_id
location current_location
